renpy.el

```diff
@@ -1529,6 +1529,109 @@ Uses `renpy-beginning-of-block', `renpy-end-of-block'."
   (renpy-end-of-block)
   (exchange-point-and-mark))
 
+;;;; Completion
+
+(defcustom renpy-setup-completion t
+  "Non-nil means Renpy mode sets up completion for the buffer."
+  :type 'boolean
+  :group 'renpy)
+
+(defun renpy--completion-context ()
+  "Return the completion category for point."
+  ;; TODO: Make sure the point is not within a comment or a string.
+  ;; TODO: More specific contexts: show text, show expression, ...
+  (save-excursion
+    (skip-syntax-backward "w_")
+    (let* ((_ (skip-syntax-backward " "))
+           (prev  (thing-at-point 'symbol)))
+      (cond
+       ;; call <point>
+       ((equal prev "call") :label)
+       ;; jump <point>
+       ((equal prev "jump") :label)
+       ;; show/scene/hide <point>
+       ((member prev '("show" "scene" "hide")) :image)
+       ;; at <point>
+       ((equal prev "at") :transform)
+       ;; default
+       (t :none)))))
+
+(defvar renpy--label-definition-re
```

Comment:
Instead of `defvar` for the regex patterns you could use `defconst` instead, to indicate that patterns should not be changed and ensure that they are always re-evaluated if the code is changed. Given the limited context, I think it would also be fine to just hard code the rx forms directly - I wouldn't really consider these for generic use if only part of the regex match is captured.

Reply:
Agree.